core/server/services/permissions/providers.js

```diff
@@ -15,6 +15,10 @@ module.exports = {
                     }));
                 }
 
+                if (foundUser.get('status') !== 'active') {
```

Comment:
Ensures redirect to `/signin` for the Ghost-Admin when the user becomes "locked" with an active session.